src/rgw/rgw_bucket.cc

```diff
@@ -255,48 +253,44 @@ static void dump_mulipart_index_results(list<rgw_obj_index_key>& objs_to_unlink,
 void check_bad_user_bucket_mapping(RGWRadosStore *store, const rgw_user& user_id,
 				   bool fix)
 {
-  RGWUserBuckets user_buckets;
-  bool is_truncated = false;
+  rgw::sal::RGWBucketList user_buckets;
+  rgw::sal::RGWRadosUser user(store, user_id);
   string marker;
 
   CephContext *cct = store->ctx();
 
   size_t max_entries = cct->_conf->rgw_list_buckets_max_chunk;
 
   do {
-    int ret = rgw_read_user_buckets(store, user_id, user_buckets, marker,
-				    string(), max_entries, false,
-				    &is_truncated);
+    int ret = user.list_buckets(marker, string(), max_entries, false, user_buckets);
     if (ret < 0) {
       ldout(store->ctx(), 0) << "failed to read user buckets: "
 			     << cpp_strerror(-ret) << dendl;
       return;
     }
 
-    map<string, RGWBucketEnt>& buckets = user_buckets.get_buckets();
-    for (map<string, RGWBucketEnt>::iterator i = buckets.begin();
+    map<string, RGWSalBucket*>& buckets = user_buckets.get_buckets();
```

Comment:
and rely on namespace qualification here, and related

Reply:
RGWBucket exists, and is needed (currenly) by the SAL code.  The plan is to rename RGWSalBucket to RGWBucket when it's possible.

Reply:
RGWSalBucket is renamed RGWBucket in the Bucket PR, to be submitted soon.